packages/core/src/auth/auth.ts

```diff
@@ -186,7 +186,14 @@ export class Auth implements AuthService, ConnectionManager {
             // We cannot easily set isReAuth inside the createToken() call,
             // so we need to set it here.
             await telemetry.aws_loginWithBrowser.run(async span => {
-                span.record({ isReAuth: true, credentialStartUrl: profile.startUrl })
+                span.record({
+                    isReAuth: true,
+                    credentialStartUrl: profile.startUrl,
+                    reAuthReason: await ReAuthReasonState.instance.getReason({
```

Comment:
this has the same risk as other global state, which e.g. codewhisperer has suffered from. 

are we certain this can't be achieved with a more direct plumbing-through of the "reauth reason", instead of setting global mutable state?